### ISSUE TITLE
Wyze Cam OG not working

### ISSUE THREAD

**Issue by mr-bobdobalina** [mrlt8/docker-wyze-bridge#677]
Not seeing the Wyze Cam OG showing up on the bridge. Running docker compose up (without the -d) I don't even see it listed in the initial "Connecting to..." section.

**Comment by ScottySTL** [comment]
I just got my new Wyze Cam OG going as well and it's not showing up on the bridge either. I am assuming it is not yet supported?

**Comment by talormanda** [comment]
Same issue here.

**Comment by jmluxem** [comment]
Same issue here. I have the new Pan v3 also, which worked right away. Wyze Cam OG is not working.

**Comment by mrlt8** [comment]
Thanks for the data points! The OG seems to be using a different P2P protocol than all the other cameras (similar to the Doorbell Pro). 
Currently looking into alternate methods to access the stream.

**Comment by ScottySTL** [comment]
Do you think it will be possible to find a way to access the stream via docker at some point in the future?

**Comment by adricrist87** [comment]
Same here, hopefully something is worked out within the 30 day return window otherwise they are going back :(

**Comment by ScottySTL** [comment]
Unfortunately, this could be Wyze purposely finding ways to now allow this kind of access to streams (So they can keep pushing their Cam Plus subscriptions)

**Comment by cheme75** [comment]
> Unfortunately, this could be Wyze purposely finding ways to now allow this kind of access to streams (So they can keep pushing their Cam Plus subscriptions)

Sadly I tend to agree it may well be wyze intentionally changing for the new hardware to force the subscription. It started to become clear when the lite version was only offered for v2 and v3 cams and the new hardware omitted the lite option. I really don’t need all the extras.  For next house I’m considering going with something like amcrest poe cams if I can get some cables run before they drywall it. Or I’ll breakdown and get a couple HomeKit compatible cams since we really won’t need much in new location.

**Comment by mrlt8** [comment]
I believe the protocol change has more to do with the fact that the OG/3X and the Doorbell Pro seem to be designed and/or produced by "gwell", whereas the other cameras were mostly Hualai cameras.

**Comment by adricrist87** [comment]
> I believe the protocol change has more to do with the fact that the OG/3X and the Doorbell Pro seem to be designed and/or produced by "gwell", whereas the other cameras were mostly Hualai cameras.

Hey thanks for getting back to us. Do you anticipate we will eventually be able to bring these OG into the bridge container? Is this work in flight or out of scope at the moment? That would help me decide wether to return these or hold on to them.

Thank you.

**Comment by mrlt8** [comment]
Potentially, but it may be more cloud dependent than our current solution (which could also mean it gets blocked by wyze).

Can you test: `http://localhost:5000/webrtc/ogcam-name`

**Comment by adricrist87** [comment]
> Tried this link http://192.168.2.101:5000/webrtc/Driveway2 and got {"cam":"Driveway2","result":"cam not found"}

**Comment by talormanda** [comment]
> > Tried this link http://192.168.2.101:5000/webrtc/Driveway2 and got {"cam":"Driveway2","result":"cam not found"} but if I try a Cam V2 or V3 I can pull them up fine.

same

**Comment by adricrist87** [comment]
Here is my docker log, It only picks up the 3 V3 Cams:

2023/01/21 03:56:01 [WyzeBridge] 🚀 STARTING DOCKER-WYZE-BRIDGE v1.11.3
 * Serving Flask app 'frontend'
 * Debug mode: off
2023/01/21 03:56:01 [WyzeBridge] 📚 Using 'user' from local cache...
2023/01/21 03:56:01 [WyzeBridge] ♻️ Refreshing camera data for thumbnails
2023/01/21 03:56:01 [WyzeBridge] 📚 Using 'auth' from local cache...
2023/01/21 03:56:01 [WyzeBridge] ☁️ Fetching 'cameras' from the Wyze API...
2023/01/21 03:56:01 [WyzeBridge] WARNING: This is a development server. Do not use it in a production deployment. Use a production WSGI server instead.
 * Running on all addresses (0.0.0.0)
 * Running on http://127.0.0.1:5000
 * Running on http://172.17.0.4:5000
2023/01/21 03:56:01 [WyzeBridge] Press CTRL+C to quit
2023/01/21 03:56:06 [WyzeBridge] 💾 Saving 'cameras' to local cache...
2023/01/21 03:56:06 [WyzeBridge] 
🎬 STARTING ALL 3 CAMERAS
2023/01/21 03:56:12 [WyzeBridge] ☁️ Pulling "Front Door Cam" thumbnail
2023/01/21 03:56:13 [WyzeBridge] 192.168.2.215 - - [21/Jan/2023 03:56:13] "GET /api/sse_status HTTP/1.1" 200 -
2023/01/21 03:56:17 [WyzeBridge] ☁️ Pulling "Driveway Cam" thumbnail
2023/01/21 03:56:23 [WyzeBridge] ☁️ Pulling "Back Yard Cam" thumbnail
2023/01/21 03:56:23 [WyzeBridge] SET WB_IP to allow WEBRTC connections.
2023/01/21 03:56:23 [WyzeBridge] Starting rtsp-simple-server v0.21.1
2023/01/21 03:56:23 [WyzeBridge] 🎉 Connecting to WyzeCam V3 - front-door-cam on 192.168.101.75 (1/3)
2023/01/21 03:56:23 [WyzeBridge] 🎉 Connecting to WyzeCam V3 - driveway-cam on 192.168.101.217 (1/3)
2023/01/21 03:56:23 [WyzeBridge] 🎉 Connecting to WyzeCam V3 - back-yard-cam on 192.168.101.234 (1/3)
2023/01/21 03:56:30 [Back Yard Cam] 📡 Getting 120kb/s HD stream (H264/20fps) via LAN mode (WiFi: 73%) FW: 4.36.9.139 🔒 (DTLS) (2/3)
2023/01/21 03:56:30 [Back Yard Cam] 🔊 Audio Enabled - ALAW/16,000Hz
2023/01/21 03:56:30 [Driveway Cam] 📡 Getting 120kb/s HD stream (H264/20fps) via LAN mode (WiFi: 36%) FW: 4.36.9.139 🔒 (DTLS) (2/3)
2023/01/21 03:56:30 [Driveway Cam] 🔊 Audio Enabled - ALAW/16,000Hz
2023/01/21 03:56:30 [Back Yard Cam] WARNING: Skipping smaller frame at start of stream (frame_size=1)
2023/01/21 03:56:30 [Driveway Cam] WARNING: Skipping smaller frame at start of stream (frame_size=1)
2023/01/21 03:56:30 [Front Door Cam] 📡 Getting 120kb/s HD stream (H264/20fps) via LAN mode (WiFi: 36%) FW: 4.36.9.139 🔒 (DTLS) (2/3)
2023/01/21 03:56:30 [Front Door Cam] 🔊 Audio Enabled - ALAW/16,000Hz
2023/01/21 03:56:30 [Front Door Cam] WARNING: Waiting for keyframe
2023/01/21 03:56:32 [RTSP][FRONT-DOOR-CAM] ✅ '/front-door-cam' stream is UP! (3/3)
2023/01/21 03:56:32 [RTSP][DRIVEWAY-CAM] ✅ '/driveway-cam' stream is UP! (3/3)
2023/01/21 03:56:33 [RTSP][BACK-YARD-CAM] ✅ '/back-yard-cam' stream is UP! (3/3)

**Comment by mrlt8** [comment]
I pushed a new commit to the dev branch. Could you try using that to see if it recognizes the OG in the bridge and lets you access the webrtc endpoint?

**Comment by jmluxem** [comment]
Am I able to test the dev branch using the Home Assistant plugin? If so, I could test right now

**Comment by mrlt8** [comment]
You should be able to access the DEV branch using this repo https://github.com/mrlt8/edge-repo

**Comment by jmluxem** [comment]
Getting the following error when trying to add the edge-repo:

Cmd('git') failed due to: exit code(128) cmdline: git clone -v --recursive --depth=1 --shallow-submodules https://github.com/mrlt8/docker-wyze-bridge/tree/dev /data/addons/git/69af740d stderr: 'Cloning into '/data/addons/git/69af740d'... fatal: repository 'https://github.com/mrlt8/docker-wyze-bridge/tree/dev/' not found '

**Comment by jmluxem** [comment]
Thanks, got the edge-repo setup in Home Assistant. Dev branch is still only showing 6 of my 7 cameras - OG is not listed.

**Comment by jmluxem** [comment]
and not seeing the OG listed anywhere in logs. Doesn't appear to be detecting it

**Comment by ScottySTL** [comment]
I tried the same with dev branch and not seeing my new OG camera listed

**Comment by adricrist87** [comment]
Is there a Discord or other chat room we can use to coordinate testing?

**Comment by camblack** [comment]
Still not detected after these latest commits. 

{"cam":"og-cam","result":"cam not found"}

```2023/01/22 22:49:37 [WyzeBridge] 🚀 STARTING DOCKER-WYZE-BRIDGE v1.11.4
2023/01/22 22:49:37 [WyzeBridge] ♻️ FORCED REFRESH - Ignoring local 'user' data
 * Serving Flask app 'frontend'
 * Debug mode: off
2023/01/22 22:49:37 [WyzeBridge] ♻️ FORCED REFRESH - Ignoring local 'auth' data
2023/01/22 22:49:37 [WyzeBridge] ☁️ Fetching 'auth' from the Wyze API...
2023/01/22 22:49:37 [WyzeBridge] WARNING: This is a development server. Do not use it in a production deployment. Use a production WSGI server instead.
 * Running on all addresses (0.0.0.0)
 * Running on http://127.0.0.1:5000
 * Running on http://192.168.1.147:5000
2023/01/22 22:49:37 [WyzeBridge] Press CTRL+C to quit
2023/01/22 22:49:38 [WyzeBridge] 💾 Saving 'auth' to local cache...
2023/01/22 22:49:38 [WyzeBridge] ☁️ Fetching 'user' from the Wyze API...
2023/01/22 22:49:38 [WyzeBridge] 💾 Saving 'user' to local cache...
2023/01/22 22:49:38 [WyzeBridge] ♻️ FORCED REFRESH - Ignoring local 'cameras' data
2023/01/22 22:49:38 [WyzeBridge] ☁️ Fetching 'cameras' from the Wyze API...
2023/01/22 22:49:39 [WyzeBridge] 💾 Saving 'cameras' to local cache...
2023/01/22 22:49:39 [WyzeBridge] 
🎬 STARTING ALL 7 CAMERAS
2023/01/22 22:49:39 [WyzeBridge] SET WB_IP to allow WEBRTC connections.
2023/01/22 22:49:39 [WyzeBridge] Starting rtsp-simple-server v0.21.1
2023/01/22 22:49:39 [WyzeBridge] 🎉 Connecting to WyzeCam Pan V2 - living-room-cam on 192.168.1.45 (1/3)
2023/01/22 22:49:39 [WyzeBridge] 🎉 Connecting to WyzeCam V3 - mom-driveway on 192.168.2.2 (1/3)
2023/01/22 22:49:39 [WyzeBridge] 🎉 Connecting to WyzeCam V3 - back-yard-cam on 192.168.1.162 (1/3)
2023/01/22 22:49:39 [WyzeBridge] 🎉 Connecting to WyzeCam V3 - mom-porch on 192.168.74.4 (1/3)
2023/01/22 22:49:39 [WyzeBridge] 🎉 Connecting to WyzeCam V3 - mom-backyard on 192.168.74.7 (1/3)
2023/01/22 22:49:39 [WyzeBridge] 🎉 Connecting to WyzeCam V3 - food-cam on 192.168.1.40 (1/3)
2023/01/22 22:49:39 [WyzeBridge] 🎉 Connecting to WyzeCam V3 - bedroom on 192.168.1.43 (1/3)```

**Comment by camblack** [comment]
On 1.11.5 it seems that it's no longer able to fetch cameras

```
2023/01/24 16:51:46 [WyzeBridge] 🚀 STARTING DOCKER-WYZE-BRIDGE v1.11.5
2023/01/24 16:51:46 [WyzeBridge] 🔍 Could not find local cache for 'user'
 * Serving Flask app 'frontend'
 * Debug mode: off
2023/01/24 16:51:46 [WyzeBridge] 🔍 Could not find local cache for 'auth'
2023/01/24 16:51:46 [WyzeBridge] WARNING: This is a development server. Do not use it in a production deployment. Use a production WSGI server instead.
 * Running on all addresses (0.0.0.0)
 * Running on http://127.0.0.1:5000
 * Running on http://192.168.1.147:5000
2023/01/24 16:51:46 [WyzeBridge] ☁️ Fetching 'auth' from the Wyze API...
2023/01/24 16:51:46 [WyzeBridge] Press CTRL+C to quit
2023/01/24 16:51:46 [WyzeBridge] 💾 Saving 'auth' to local cache...
2023/01/24 16:51:46 [WyzeBridge] ☁️ Fetching 'user' from the Wyze API...
2023/01/24 16:51:47 [WyzeBridge] 💾 Saving 'user' to local cache...
2023/01/24 16:51:47 [WyzeBridge] 🔍 Could not find local cache for 'cameras'
2023/01/24 16:51:47 [WyzeBridge] ☁️ Fetching 'cameras' from the Wyze API...
2023/01/24 16:51:48 [WyzeBridge] 1 validation error for WyzeCamera
ip
  value is not a valid integer (type=type_error.integer)

**Comment by jmluxem** [comment]
Wyze Cam OG is being detected in 1.11.6!
It shows a recent snapshot but it isn't able to load the WebRTC stream - gives a 404 error for both WebRTC and HLS.
All other cams working normally.

**Comment by sguy311** [comment]
So, the OG is at least showing up on the page, but here is the log entry about it
2023/01/24 21:32:40 [WyzeBridge] 🎉 Connecting to WyzeCam OG - carport on  (1/3)
2023/01/24 21:32:40 [Carport ] IOTC_ER_UNLICENSE

**Comment by mrlt8** [comment]
Type validation should be fixed in v1.11.6.

IOTC_ER_UNLICENSE is probably because the camera isn't registered with tutk when it tries to look for it remotely. It may be using Tencent's IoTVideoSdk like the doorbell pro (#276) for NAT punching/p2p.

can someone try the `:5000/webrtc/<og-cam-name>` endpoint?

**Comment by mrlt8** [comment]
Made some tweaks to avoid `IOTC_ER_UNLICENSE` and swapped back to the default api.

**Comment by ScottySTL** [comment]
Dev or Latest branch?

**Comment by mrlt8** [comment]
Dev branch

**Comment by ScottySTL** [comment]
Got it, now the OG camera doesn't show up detected in list anymore

**Comment by mrlt8** [comment]
It should still be there, the bridge just won't attempt to connect to it directly since they're using a different p2p protocol.

Do you get anything on `:5000/webrtc/<cam-name>`?

**Comment by camblack** [comment]
The OG cameras are detected now on my end, however I do get a "Not supported" error for one of them in the log. 
WebRTC endpoint: {"cam":"og-cam","result":"Camera does not support WebRTC"} 

Interesting that one of the OG cams (Litter Cam) shows up as "Not Supported" while the other OG cam shows up as connecting before being removed. 

```
2023/01/25 04:49:00 [WyzeBridge] 💔 Litter Cam [GW_GC1] is not supported
2023/01/25 04:49:00 [WyzeBridge] 
🎬 STARTING ALL 8 CAMERAS
2023/01/25 04:49:00 [WyzeBridge] SET WB_IP to allow WEBRTC connections.
2023/01/25 04:49:00 [WyzeBridge] Starting rtsp-simple-server v0.21.1
2023/01/25 04:49:01 [WyzeBridge] 🎉 Connecting to WyzeCam OG - og-cam on  (1/3)
2023/01/25 04:49:01 [WyzeBridge] 🎉 Connecting to WyzeCam Pan V2 - living-room-cam on 192.168.1.45 (1/3)
2023/01/25 04:49:01 [WyzeBridge] 🎉 Connecting to WyzeCam V3 - mom-driveway on 192.168.2.2 (1/3)
2023/01/25 04:49:01 [WyzeBridge] 🎉 Connecting to WyzeCam V3 - back-yard-cam on 192.168.1.162 (1/3)
2023/01/25 04:49:01 [WyzeBridge] 🎉 Connecting to WyzeCam V3 - mom-porch on 192.168.74.4 (1/3)
2023/01/25 04:49:01 [WyzeBridge] 🎉 Connecting to WyzeCam V3 - mom-backyard on 192.168.74.7 (1/3)
2023/01/25 04:49:01 [WyzeBridge] 🎉 Connecting to WyzeCam V3 - food-cam on 192.168.1.40 (1/3)
2023/01/25 04:49:01 [WyzeBridge] 🎉 Connecting to WyzeCam V3 - bedroom on 192.168.1.43 (1/3)
2023/01/25 04:49:01 [OG Cam] IOTC_ER_UNLICENSE
2023/01/25 04:49:02 [WyzeBridge] Removing camera

**Comment by mrlt8** [comment]
hmm, that's weird. Not sure why only some cams are missing some fields..

Updated the dev branch so that it doesn't remove the weird `GW_GC1_` prefix from the Mac address.

**Comment by warmon6** [comment]
Throwing my OG telephoto in the ring. 
As predicted, same as the others with the regular OG's
(This is on the dev branch)
```
2023/01/26 20:08:25 [WyzeBridge] 💔 street cam [GW_GC2] is not supported
2023/01/26 20:08:25 [WyzeBridge] 
🎬 STARTING ALL 2 CAMERAS
2023/01/26 20:08:25 [WyzeBridge] ☁️ Pulling "Front Porch Cam" thumbnail
2023/01/26 20:08:26 [WyzeBridge] ☁️ Pulling "Driveway Cam" thumbnail
2023/01/26 20:08:26 [WyzeBridge] Starting rtsp-simple-server v0.21.1
2023/01/26 20:08:26 [WyzeBridge] [ON-DEMAND] ⌛️ WyzeCam V3 - front-porch-cam on 192.168.50.56 (1/3)
2023/01/26 20:08:26 [WyzeBridge] [ON-DEMAND] ⌛️ WyzeCam V3 - driveway-cam on 192.168.50.96 (1/3)
```

**Comment by mrlt8** [comment]
Thanks for the confirmation @warmon6!

Has anyone been able to get the OG to show up in the official web view or is the OG not supported there?

**Comment by camblack** [comment]
I hadn't even bothered to check until now, but they do indeed show up and play in the official web view!

**Comment by talormanda** [comment]
> Thanks for the confirmation @warmon6!
> 
> Has anyone been able to get the OG to show up in the official web view or is the OG not supported there?

Is there any data you need out of the official webview that would help? Do you need an OG camera to test? I could probably set up one and share it with you if you think that would help get it working.

**Comment by ramboton** [comment]
> Thanks for the confirmation @warmon6!
> 
> Has anyone been able to get the OG to show up in the official web view or is the OG not supported there?

My OG works on webview

**Comment by cryptosmasher** [comment]
out of absolutely sheer docker ignorance, how do I pull from the edge-repo on the command line via docker run ? (OG/OG 3X/V3 Pan user willing to test) thanks

**Comment by sanderz19** [comment]
First off, the fact that this add-on exists and works is amazing. So thank you for that. I am running HAOS, and I have a number of Wyze cameras connected (2 of mine and 4 that are shared) and working beautifully. I also have 3 new OG cameras (1 OG and 1 OG Telephoto locally, and another shared OG). The three OGs are not working. I am running the latest update that was pushed today (v1.11.8/9/10). Is there anything I can do to test or contribute?

**Comment by jcalderon71** [comment]
Same issue, running Docker Wyze bridge v 1.11.10 on home assistant 2023.2.3 with 4 Wyze v3 working fine and 4 Wyze OG not working. Also getting a lot stop reading  issues.
2023/02/10 20:38:01 [RTSP][BABY-CAM] 📕 Client stopped reading
2023/02/10 20:38:03 [RTSP][BACK-YARD-CAM] 📕 Client stopped reading
2023/02/10 20:38:04 [RTSP][OFFICE-CAM] 📕 Client stopped reading
2023/02/10 20:38:06 [RTSP][BABY-CAM] 📖 New client reading 
2023/02/10 20:38:08 [RTSP][BACK-YARD-CAM-2] 📖 New client reading 
2023/02/10 20:38:09 [RTSP][OFFICE-CAM] 📖 New client reading 
2023/02/10 20:38:10 [RTSP][BACK-YARD-CAM-2] 📕 Client stopped reading
2023/02/10 20:38:11 [RTSP][BABY-CAM] 📕 Client stopped reading
2023/02/10 20:38:14 [RTSP][OFFICE-CAM] 📕 Client stopped reading

**Comment by TEfjLONjDON** [comment]
First off this is an awesome project so far..  Very helpful. After 2 weeks of hoping the dev branch would get updated again and perhaps I would start working I've moved back to normal branch . But I'd be happy to test.  I'd truly love to get these OG cams markers as supported. 

Also; When I try ```    :5000/webrtc/<og-cam-name>    ``` 
It loads the player view. It circles for a while and act as if it might load the feed But ultimately it never loads The camera stream.

**Comment by mach7** [comment]
@mrlt8 do you have any update on this? If you'd like help with getting information for this camera I'd be willing to purchase one and help out with it. Depending on where you're located and shipping, I may be willing to purchase one for each of us as a contribution to the project.

**Comment by PJW58** [comment]
I just updated my Bridge to 2.x and noticed that the api snapshot is now working for my regular OG.  However it gives a 404 for my OG Telephoto.  Video streams still not working for either.

**Comment by Mdleal** [comment]
Running 2.0. Getting the following. Stream does not work. Thumbnail does but its the last motion event. 

[WyzeBridge] 🎉 Connecting to WyzeCam OG - Pet Cam on 
[pet-cam] [-10] IOTC_ER_UNLICENSE
[WyzeBridge] Pet Cam will cooldown for 10s.

**Comment by mrlt8** [comment]
OG/Doorbell Pro are not supported at this time.

**Comment by drkavnger99** [comment]
Wondering if the OG's are being actively pursued to add or not. Trying to figure out if I should bother setting them up or not.

**Comment by jhansche** [comment]
I have some HTTP proxy logs, and limited wireshark logs, with the ability to capture additional traffic between the Wyze app and the OG cam. How can I help with researching or adding support?

I'm willing to contribute changes, if someone can point me in the right direction for building and testing changes.

**Comment by mrlt8** [comment]
The OG is most likely using a different "IoTVideoSdk" platform. The SDK seems to be from Gwell/Tencent like the doorbell pro which @carTloyal123 was looking into.

Unfortunately, there isn't much information or any documentation in English:
https://github.com/GWTimes/IoTVideo-PC
https://github.com/GWTimes/GWP2PSDK
https://cloud.tencent.com/document/product/1131/42244

**Comment by jhansche** [comment]
Isn't it this? https://www.tencentcloud.com/document/product/494/7244

All we need the tencent API for is to obtain the access key (enr) and get the local IP and UDP port, is that right? The wyze API response shows only a "placeholder" for the OG camera enr, and no ip.

Once we have the IP, port, and enr (is this the same as tencent's access key?), then there's a small handshake negotiation that happens between device and app, and after that the video starts to stream over UDP. This is presumably where the GWP2P or the iotvideo SDK comes in.

https://github.com/GWTimes/IoTVideo-Android looks a bit more informative.

So, high level, how would you describe what needs to happen in order to make progress towards support?
* How much of it _is_ shared between GW and other wyze cams? E.g. it seems you already have "IOTC" in the stack somewhere, so that's not new.. we can get the thumbnails, so that's not new, etc. Where exactly is the separation between them?
* once we have the access key for the camera, is that all that's missing? Or do we still need a way to communicate with the camera P2P (the UDP handshake and streaming mentioned above)?

I just want to make sure I'm working in the right things, and not chasing something that's irrelevant or already in place.

@cartloyal123 anything new to share, or roadblocks you've hit that I can help get past?

**Comment by spartandrew18** [comment]
Any progress on getting the Wyze OG to work?

**Comment by sdalbert** [comment]
I also was happy to see progress for the OG recently and would like to see more. I understand that it is not promising so far, but I'm invested in one of these now and would rather not send it along to the obsolete hardware box yet.

**Comment by iamdagy** [comment]
This is a very useful and needed project. Thanks a lot. Hope OG cameras are Supported soon (I've got a bunch of them) your bridge solution can finally free the feed from a specific webui/app.
Let us know how we can help in getting Support for OG cameras

**Comment by neo8** [comment]
Just FYI, I upgraded to 223 and partially connected. I see a still image with "The media could not be loaded, either because the server or network failed because the format is not supported. X" My other V3's and V2's work fine.

**Comment by iamdagy** [comment]
Is there anyway we can help moving forward the support for the new OG camera ?

**Comment by carTloyal123** [comment]
Hi everyone, sorry for the radio silence. I have not looked much further into the new cameras since I posted a while ago. The python Tencent library looks very promising and I had not come across that. If someone wants to give that a shot that would be awesome but I will not be able to give this much time for the next several weeks/months sadly. I was working on this to get a stream from the video doorbell pro and basically was stuck at trying to reverse engineer the mechanism to even enable the camera video stream to start. I hope to have time to look at these at some point but for now I will just be in the background/checking this thread.

**Comment by nbetcher** [comment]
> I also was happy to see progress for the OG recently and would like to see more.
> Just FYI, I upgraded to 223 and partially connected. I see a still image with "The media could not be loaded, either because the server or network failed because the format is not supported. X" 
> Is there anyway we can help moving forward the support for the new OG camera ? 

Where is it that people are saying this, "support" and "progress"? I'll be honest, I don't follow this project's commits closely and since they're made often but from what I could tell, there is no progress. And that would follow what I understand about the original problem. Which leads to my response to these replies to:

> I was working on this to get a stream from the video doorbell pro and basically was stuck at trying to reverse engineer the mechanism to even enable the camera video stream to start.
> What I've seen so far in my wire dumps is that the wyze api does not send the actual encryption key or the local ip address for these cameras, like it does for the other cameras

My understanding is that @mrlt8 is not interested in reverse engineering or implementing an implementation of a reversed-engineered solution (totally speculation). I say this only because the currently-supported cameras are implemented by Python library derived directly from a *leaked* copy of the library from the ThroughTek (Tutk) IoT SDK [1].

These new cameras do not use the Tutk IoT SDK and instead use Gwell Times. Most of you know this already, but going back to what I originally said...

It makes me think that, unless someone comes up with the library from the original Gwell Times SDK (which I believe is floating around on Github, maybe?), *AND* an SDK key, *AND* @mrlt8 is willing to spend all of that time implementing another SDK (take a look at the amazing work he's done already in https://github.com/mrlt8/docker-wyze-bridge/tree/25a2bdfa2ad7f992de4db5cac55923fbb68779a6/app/wyzecam and that's despite the massive, massive amounts of heavy-lifting that the Tutk SDK's native library already handles.

I don't know how easily getting an SDK key will be though. That might be the tougher of the two things to accomplish.

Nothing I've said here about @mrlt8 represents any degree of accuracy. I'm not a developer or contributor on this project. I am a software developer, but I don't have the time or A/V / mathematical knowledge to accomplish the end result. These are just my impressions and interpretation since @mrlt8 has been a bit quiet on the topic.


[1] See https://github.com/mrlt8/docker-wyze-bridge/blob/25a2bdfa2ad7f992de4db5cac55923fbb68779a6/app/wyzecam/iotc.py#L55

**Comment by mrlt8** [comment]
While I was a huge fan of the original v2 and v3 cams (and still own more cams than I actually need), I have little interest/need for any of the gwell cameras. These newer models are very cloud dependent and have a higher chance of becoming ewaste if wyze drops support and/or goes under. 

However, I am still working on an alternate way to add support for the newer cameras - the floodlight pro and the unreleased doorbell/outdoor cam also seem to be from yet another manufacturer so I'd like to find a solution that could potentially work across multiple platforms.

As for the gwell related credentials, has anyone looked into 
`https://wyze-mars-service.wyzecam.com/plugin/mars/v2/regist_gw_user/`? 
I can provide the signature keys for the relevant app-ids if needed.

**Comment by gibsonMatt** [comment]
my Micro Center is selling these for $15..picked up a few. How are things looking for eventual support of gwell cameras?

**Comment by gtxaspec** [comment]
@mrlt8 i think the common platform with the new cameras even between the different mfg's is the Realtek Ameba Pro 2  IOT Platform:   
https://www.amebaiot.com/en/amebapro2  
https://github.com/ambiot/ambz2_sdk  
https://github.com/aws-samples/amazon-kinesis-video-streams-producer-embedded-c

Also looks like the tiny cam pro developer has figured out support too.  may be worthwhile decompiling that app.

**Comment by mrlt8** [comment]
Thanks! Will try to take a look when I get a chance!

**Comment by nbetcher** [comment]
> Thanks! Will try to take a look when I get a chance!

I've done some very preliminary inspection of how TInycam is accomplishing this and I believe the developer applied for, and received, the Tencent SDK to interface with the Gwell cameras. The Tencent SDK is definitely inside of Tinycam, along with some heavy mathematical obfuscation methods, likely to mask the developer's SDK key.

Some interesting strings:

https://wyze-mars-service.wyzecam.com/
wyze-mars-asrv.wyzecam.com
wyze-test-list.cloudlinks.cn

The rest of the class is very, very, VERY heavily obfuscated, and even AI-based decompilation would struggle given the degree of nesting.

Either way, it appears the developer went the legitimate route and just obtained an SDK and a key for it.

**Comment by a9s2w5** [comment]
It should also be noted that even TinyCam's implementation only runs/supports the new OG/Doorbell Pro on ARM devices. There is no x86 support. And the TinyCam DEV had been hired to actually do work for Wyze, not sure what or the end result of that. This is me asking as much as suggesting, it seems like support for the Gwell/Tencent products, such as the OG and Doorbell Pro, are highly unlikely to be supported due to the complications/limitations?

**Comment by talormanda** [comment]
> It should also be noted that even TinyCam's implementation only runs/supports the new OG/Doorbell Pro on ARM devices. There is no x86 support. And the TinyCam DEV had been hired to actually do work for Wyze, not sure what or the end result of that. This is me asking as much as suggesting, it seems like support for the Gwell/Tencent products, such as the OG and Doorbell Pro, are highly unlikely to be supported due to the complications/limitations?

He got hired by wyze? where did you read that?

**Comment by a9s2w5** [comment]
> He got hired by wyze? where did you read that?

https://www.linkedin.com/in/avasilyev/

**Comment by ramboton** [comment]
I am not so sure about the no x86 support either. I am viewing my doorbell cam with TinyCam pro, I am on a chromebook, but it does work and my chromebook is not ARM, it is an Intel I5-8350U

![Screenshot 2023-08-09 6 50 11 PM](https://github.com/mrlt8/docker-wyze-bridge/assets/79012966/566f656e-5a70-48c4-aad1-2fcb0e341cd9)
![Screenshot 2023-08-09 6 50 51 PM](https://github.com/mrlt8/docker-wyze-bridge/assets/79012966/ac110816-abc5-44fb-8387-84ec55da7239)
 I5-8350.

**Comment by a9s2w5** [comment]
> I am not so sure about the no x86 support either. I am viewing my doorbell cam with TinyCam pro, I am on a chromebook, but it does work and my chromebook is not ARM, it is an Intel I5-8350U
> 

Merely am quoting the developer of Tinycam. https://www.reddit.com/r/tinycam/comments/159x0gg/comment/juuy0di/?utm_source=share&utm_medium=web2x&context=3

**Comment by iamdagy** [comment]
have we got any news ? how can we help to move this forward? Thanks

**Comment by carTloyal123** [comment]
It sounds like to me there are two ways to actually push this forward. The first would be for someone to reach out to Wyze/Alexey and see if we can cooperate and be on the same team here. The second would be for someone to spend a decent amount of time trying to implement the Realtek SDK and/or the Tencent SDK into the bridge which will seemingly be quite difficult given many unknowns about the credentials and auth management needed. I am happy to do the first and see if we can get anywhere by gaining knowledge from Alexey and Tinycam. If that does not go anywhere then people can speak up on what they know or how much they can contribute to getting anywhere with adding the new SDKs and figuring out the auth steps that will be needed. @mrlt8 how does this sound to you?

**Comment by nbetcher** [comment]
Or a third option: just ask Alexey how the authentication and credentialing is to see if the second option is more practical.

**Comment by carTloyal123** [comment]
@nbetcher Sure, but this is still basically the first option. 

I have reached out to Alexey on LinkedIn to see what insight he is willing to provide.

**Comment by carTloyal123** [comment]
An update: I have successfully backed out the wyze information needed to get the p2p accessId and accessToken and this is working in python. With these two pieces of information along with a deviceId, p2purl (known) and the aformentioned iotvideo SDKs we might get somewhere. 

At the moment, I am trying to load up the iotvideo sdk in python similar to the tutk library that is already implemented. This is where I could use some help probably from @mrlt8 since my knowledge is being stretched. I have the .so files for the libraries but I can not call any functions by name after loading them with ctypes CDLL. I am curious if you could offer some insight or if you would be interested in taking a stab at it yourself. 

Regardless, half of the equation is pretty much done which is getting the credentials from Wyze so that is progress. I will keep pushing on getting these libraries to play nice in python but until then happy coding!

**Comment by jhansche** [comment]
@carTloyal123 that's awesome 😎

* Do you have your WIP committed somewhere to peruse or experiment with?
* Where did the .so file come from? (i.e., extracted from an android APK, iOS app, some other backchannel, etc? More directly: what platform was it intended for?
* If the .so file was not actually written for Python's C API you'll need to do some wrapping, in order to act as a bridge between the Python runtime and the C library, so that the Python runtime can call the wrapper, which will import this C library and call the target code in that library. Have you already done this? See https://docs.python.org/3/extending/extending.html

**Comment by carTloyal123** [comment]
Not committed anywhere yet, I will put it in a PR for this repo soon. Libraries are currently from the wyze app apk on android and since they only support ARM platforms it seems like they do not like to be loaded. I have been trying using an aarch64 machine but still no luck. The issue is not being able to wrap them, that is trivial, the issue is working out how to even execute the binaries on a platform other than android. There is a windows dll which might be useful but I have not tried it. Since Tutk has an x86_64 version it works fine but that is not the case here for the iotvideo sdk

**Comment by jhansche** [comment]
Yes for sure, an arm or other .so file will not load on an x86 or other architecture. But Android re-bundles APKs based on the device that is installing the app. So if you installed it from an arm device, you would only get the arm libraries.

Best case would be to install it on an x86 or x86_64 device, and extract the so files from there. Then it should be able to load on a target system assuming the arch matches (or is compatible).

On my device for example, I have:
```
/data/app/~~UaTVuHj6tdXhp4vEFFTVCQ==/com.hualai-pN8RQUbB4pQPzWA8Lyj66w==/:
-rw-r--r-- 1 system system 105254430 2023-08-18 05:10 split_config.arm64_v8a.apk
```
But installing on another arch should provide other arch splits.

**Comment by carTloyal123** [comment]
As far as I can tell that actually is not the case. Each apk comes with all the needed binaries for every architecture. At runtime the developer has to check the current platform ABI and load the desired library if available. You can see this quite clearly by decompiling any apk. There are folders for each architecture type and browsing the decompiled code shows system calls to check ABI and selectively load binaries.

Now this brings me to the really interesting finding that someone on this thread is running tinycam on their x86 Chromebook and viewing their doorbell. @ramboton would you be able to do some digging by chance on this? I'm not super familiar with what android does to install apps but there has to be a place to store all the libraries somewhere and if you wanted to see if you can find that that would be awesome. You're looking for a folder called 'x86' or 'x86_64' and the file will be called 'libiotvideo.so'. I am guessing that the Chromebook has an emulation layer but hopefully I am wrong and we can magically find the x86 files we need.

**Comment by ramboton** [comment]
Unfortunately I have 2 issues, #1 is that I am not technical enough to do
that I know nothing about programming or anything like that. #2 for some
reason the OG is no longer working in Tinycam on my x86 chromebook, some
update must have broken that feature.

**Comment by jhansche** [comment]
I've been an android developer for over 10 years. The file I showed listed above is named "split" because it is from an "APK split", which works exactly as I described, and it is very well documented. If the app was published as an APK initially by the developer, then yes it must include all architectures it will support, but that hasn't been the best or most common practice for many years. If the developer publishes the app to the play console as an AAB (Android app bundle), then the play store will automatically deconstruct it and create all the splits needed and will deliver only what is relevant to your device when you install it.

That said, the AAB that was uploaded must have the architecture support to begin with, so if they never intended to support an architecture, play store won't magically create it for you. If it can be installed on an x86 device (i.e. an emulator would be my first suggestion), then the split will be present there.

**Comment by carTloyal123** [comment]
This is all fine. Either way the issue still remains that we need a way to load the libiotvideo.so library that comes with Wyze or TimyCam. I assume this is possible since @mrlt8 has been able to do the same thing for the TUTK binary. It also looks like @mrlt8 has combined the tutk and iotcapi binaries into one so I would definitely like to know more about that if possible. So far the iotvideo sdk requires at least 5 binaries so it would be great to only end up with one. @mrlt8 did you have to do anything special to get elf headers corrected from the tutk binary?

**Comment by jhansche** [comment]
It was my understanding that @mrlt8 obtained the C library from some other undisclosed source, I do not believe it came from the mobile apps, but from some other private or internal source. 🤔 but now I can't find the reference to that information to backup that notion.

The amd.lib and arm.lib files in the docker app dir are described as:
```
amd.lib: ELF 64-bit LSB shared object, x86-64, version 1 (SYSV), dynamically linked, BuildID[sha1]=8dc56493821b4c11df52c05f77bedd7d12339ef5, not stripped
arm.lib: ELF 32-bit LSB shared object, ARM, EABI5 version 1 (SYSV), dynamically linked, BuildID[sha1]=0fa09d0d94da421c35c658683d7a3d7807e422af, not stripped
```
So this being the case, it's possible that your ARM lib might work on a raspberry pi, or other ARM based host? But for it to work on an intel or AMD based architecture, we'll need to find another source for x86, x86_64, or amd64 libs.

It doesn't appear that iOS apps would contain an x86 binary either, unless the app is intended to run on the Simulator. Otherwise it's all ARM based. If there were a macOS app, that would surely support x86, but I do not see one.

**Comment by mrlt8** [comment]
The tutk library that we use comes from the SDK which is compiled for linux: 

>  Library files (.\Lib\)
>  To provide the greatest portability, TUTK IOTC platform supports many popular platforms and corresponding development environment. The corresponding libraries are listed as following for use when linking executable files.


Platform | Directory
-- | --
Android | .\Lib\Android
Google Native Client | .\Lib\NaCl
iOS | .\Lib\iOS
Linux (Embedded Linux) | .\Lib\Linux\<SOC>
MAC OS | .\Lib\MAC
Windows | .\Lib\Windows
WinRT | .\Lib\WniRT


https://www.sunipcam.com/sdk/Readme.htm


I don't believe it's possible to use the andorid (bionic) library directly in linux and we'd need to run some kind of emulation or convert it to glibc with something like https://github.com/Cloudef/android2gnulinux

The iOS libraries may have potential to run natively on MacOS with the ma1/m2, but not sure if it could be done in a docker container?

**Comment by carTloyal123** [comment]
Thanks @mrlt8 for the update. That is most helpful since I think I was assuming the binary had come from an android/ios app. 

Looks like android2gnulinux might be an option. At the moment I am pretty fascinated by https://github.com/libhybris/libhybris/tree/master/hybris so I am going to see if I can get a basic example of this working. From what I can tell this should be a decent way to generate C bindings, then wrap in Python when needed. It will be a long process but does at least offer some hope that this is possible. Let me know if anything else comes up or other comments!

**Comment by nbetcher** [comment]
Apologies for falling behind in the conversation — and although I did skim through all replies since my last, but was unable to answer this question — I'm confused though, why are we bothering with the TutK SDK (which is already fully impenetrable and in this project) when the it's the Tenscent SDK that is used for the Wyzecam OG/etc?

**Comment by carTloyal123** [comment]
Tutk SDK is of course done. We are only talking about the iotvideo.so library found in the GWell repos and in the wyze/tinycam apks

**Comment by ju5t3nc4s3** [comment]
the SOC in the OG is the same as this dev board (AMB82-MINI) 
  doc is here  https://device.report/m/f97fabeccb918a1d2e6c4183aeca88e7ba92ae3eff1dec7f4bbc47ef76dba633_optim.pdf

**Comment by gtxaspec** [comment]
Not an expert in audio video processing but I'm assuming that both the audio and the video have timestamps is there a possibility to buffer them individually and then once we have a buffer built up compare this timestamps when we are combining them into one stream and adjust the bufferd content to compensate for a delay in either the video or the audio?

**Comment by mrlt8** [comment]
The raw h264 bytes don't contain any PTS/DTS/timestamps. The tutk AV module does include the timestamp to the micro second with each frame, but I don't believe it's possible to pass that to ffmpeg? 

I believe the current sync issue seems to come from the mismatch in the number of video/audio frames if/when the camera drops a frame or is slow. When that happens, the muxer seems to pause the audio till it receives the next video frame causing the audio to lag behind or, if using the wallclock to set the timestamps, the audio will continue while the video slowly drifts behind as the camera starts producing frames and muxer doesn't seem to realize the video is behind since it's using the wallclock to set the timestamp as they arrive..

**Comment by revans23** [comment]
Just chiming in to voice support for an OG solution.  I have both a M2 Mac and PC that I am happy to test on if it is helpful.

**Comment by mrlt8** [comment]
Sorry for the delay, hope to have something up for testing soon.

**Comment by carTloyal123** [comment]
Amazing, is this something to test for the doorbell pro and cam og?

**Comment by carTloyal123** [comment]
@mrlt8 anything others can help with?

**Comment by dinan5** [comment]
I'm happy to test with OG Cam as well.  Thanks!

**Comment by mrlt8** [comment]
Will need to look into it, but I believe the  doorbell pro/pro 2 require a wake command so I don't think they'll be supported.

**Comment by carTloyal123** [comment]
@mrlt8 that makes sense. Are you using the iotvideo sdk for the og cam or some other approach to interface with it?

**Comment by mrlt8** [comment]
I'm looking into the native kvs. 

The display on my laptop seems to have died (flexgate?) so I'm a little behind on things.

**Comment by carTloyal123** [comment]
Are you referring to Amazon Kinesis Video Streams? Have you seen some indications that this is the cloud platform behind these cameras? Just want to help and look in the right direction if possible here! I am mainly interested in the video doorbell pro so I'll start poking around for how to wake the thing/initiate video streams. Thanks for staying on this! @mrlt8

**Comment by Simcha-Orenbuch** [comment]
Just got two wyze OG's I was planning on using for important security purposes around my house and found both Home Assistant and this docker and thought, "wow, thats awesome". 
First off, you guys are freaking geniuses, where I'm from I'm considered tech savvy but you guys are on another level.
I'm somewhat new to the in depth raspberry pi space, used it for jobs but never went that deep.
I followed all the instructions and got to the point mentioned in the thread about HA recognizing the camera, giving a thumbnail, but not supporting any of the streams. 
I have multiple raspberry pis and a few computers I can use for testing, although I'm not as technically skilled as pretty much any of you guys so idk how much help that would be.

**Comment by industrial64** [comment]
Thanks @mrlt8 for the updates, and for the bridge, without it I wouldn't have so many of their cameras. 
Hoping to get these OG and OG Telephoto's functional soon :)

**Comment by carTloyal123** [comment]
@mrlt8 any update on this? I am working on an android app that uses the Tencent iotvideo sdk that tinycam and the Wyze app use to be a shim app for getting video data. This is going to be quite a bit of work it seems so wanted to see if there is progress in any other direction I should be aware of here?

**Comment by cybersholt** [comment]
Just ran into this issue myself after getting a defective v3, I got and extra OG cam. 

Testing on Windows 10 Pro 22H2 with Docker Desktop 4.26.1 (131620) 
- HLS Video works in browser
- RTMP works with VLC
- RTSP works with VLC
- Snapshot works
- Thumbnail works 

Is there a way to disable the audio completely in the mean time to get this working? As I'm not really concerned with recording audio.

Also @carTloyal123, an app sounds great, personally I'm using [Frigate](https://frigate.video/) and [Frigate Viewer](https://github.com/sp-engineering/frigate-viewer) ( no affiliation and only Android ). Basically I'm using this project to generate the streams needed.

And finally to @mrlt8 Awesome project keep up the great work! 

`2023-12-20 12:15:53 🚀 DOCKER-WYZE-BRIDGE v2.6.0 
2023-12-20 12:15:53 
2023-12-20 12:15:53 [WyzeBridge] ♻️ Clearing local cache...
2023-12-20 12:15:53 [WyzeBridge] 🔍 Could not find local cache for 'auth'
2023-12-20 12:15:53 [WyzeBridge] ☁️ Fetching 'auth' from the Wyze API...
2023-12-20 12:15:53  * Serving Flask app 'frontend'
2023-12-20 12:15:53  * Debug mode: off
2023-12-20 12:15:53 [WyzeBridge] WARNING: This is a development server. Do not use it in a production deployment. Use a production WSGI server instead.
2023-12-20 12:15:53  * Running on all addresses (0.0.0.0)
2023-12-20 12:15:53  * Running on http://127.0.0.1:5000
2023-12-20 12:15:53  * Running on http://172.18.0.5:5000
2023-12-20 12:15:53 [WyzeBridge] Press CTRL+C to quit
2023-12-20 12:15:54 [WyzeBridge] 💾 Saving 'auth' to local cache...
2023-12-20 12:15:54 [WyzeBridge] 🔍 Could not find local cache for 'user'
2023-12-20 12:15:54 [WyzeBridge] ☁️ Fetching 'user' from the Wyze API...
2023-12-20 12:15:54 [WyzeBridge] 💾 Saving 'user' to local cache...
2023-12-20 12:15:54 [WyzeBridge] 🔍 Could not find local cache for 'cameras'
2023-12-20 12:15:54 [WyzeBridge] ☁️ Fetching 'cameras' from the Wyze API...
2023-12-20 12:15:55 [WyzeBridge] [API] Fetched [5] cameras
2023-12-20 12:15:55 [WyzeBridge] 💾 Saving 'cameras' to local cache...
2023-12-20 12:15:55 [WyzeBridge] [+] Adding Pet Cam [WYZE_CAKP2JFUS]
2023-12-20 12:15:55 [WyzeBridge] [+] Adding Back Yard Cam [GW_GC1]
2023-12-20 12:15:55 [WyzeBridge] [GW_GC1] Back Yard Cam not supported
2023-12-20 12:15:55 [WyzeBridge] [+] Adding Driveway Entrance [WYZE_CAKP2JFUS]
2023-12-20 12:15:55 [WyzeBridge] [+] Adding Driveway North  [WYZEC1-JZ]
2023-12-20 12:15:55 [WyzeBridge] [+] Adding Front Door [WYZEC1-JZ]`

**Comment by Simcha-Orenbuch** [comment]
> Just ran into this issue myself after getting a defective v3, I got and extra OG cam.
> 
> Testing on Windows 10 Pro 22H2 with Docker Desktop 4.26.1 (131620)
> 
> * HLS Video works in browser
> * RTMP works with VLC
> * RTSP works with VLC
> * Snapshot works
> * Thumbnail works
> 
> Is there a way to disable the audio completely in the mean time to get this working? As I'm not really concerned with recording audio.
> 
> Also @carTloyal123, an app sounds great, personally I'm using [Frigate](https://frigate.video/) and [Frigate Viewer](https://github.com/sp-engineering/frigate-viewer) ( no affiliation and only Android ). Basically I'm using this project to generate the streams needed.
> 
> And finally to @mrlt8 Awesome project keep up the great work!
> 
> ```
> 2023-12-20 12:15:53 
> 2023-12-20 12:15:53 [WyzeBridge] ♻️ Clearing local cache...
> 2023-12-20 12:15:53 [WyzeBridge] 🔍 Could not find local cache for 'auth'
> 2023-12-20 12:15:53 [WyzeBridge] ☁️ Fetching 'auth' from the Wyze API...
> 2023-12-20 12:15:53  * Serving Flask app 'frontend'
> 2023-12-20 12:15:53  * Debug mode: off
> 2023-12-20 12:15:53 [WyzeBridge] WARNING: This is a development server. Do not use it in a production deployment. Use a production WSGI server instead.
> 2023-12-20 12:15:53  * Running on all addresses (0.0.0.0)
> 2023-12-20 12:15:53  * Running on http://127.0.0.1:5000
> 2023-12-20 12:15:53  * Running on http://172.18.0.5:5000
> 2023-12-20 12:15:53 [WyzeBridge] Press CTRL+C to quit
> 2023-12-20 12:15:54 [WyzeBridge] 💾 Saving 'auth' to local cache...
> 2023-12-20 12:15:54 [WyzeBridge] 🔍 Could not find local cache for 'user'
> 2023-12-20 12:15:54 [WyzeBridge] ☁️ Fetching 'user' from the Wyze API...
> 2023-12-20 12:15:54 [WyzeBridge] 💾 Saving 'user' to local cache...
> 2023-12-20 12:15:54 [WyzeBridge] 🔍 Could not find local cache for 'cameras'
> 2023-12-20 12:15:54 [WyzeBridge] ☁️ Fetching 'cameras' from the Wyze API...
> 2023-12-20 12:15:55 [WyzeBridge] [API] Fetched [5] cameras
> 2023-12-20 12:15:55 [WyzeBridge] 💾 Saving 'cameras' to local cache...
> 2023-12-20 12:15:55 [WyzeBridge] [+] Adding Pet Cam [WYZE_CAKP2JFUS]
> 2023-12-20 12:15:55 [WyzeBridge] [+] Adding Back Yard Cam [GW_GC1]
> 2023-12-20 12:15:55 [WyzeBridge] [GW_GC1] Back Yard Cam not supported
> 2023-12-20 12:15:55 [WyzeBridge] [+] Adding Driveway Entrance [WYZE_CAKP2JFUS]
> 2023-12-20 12:15:55 [WyzeBridge] [+] Adding Driveway North  [WYZEC1-JZ]
> 2023-12-20 12:15:55 [WyzeBridge] [+] Adding Front Door [WYZEC1-JZ]```
> ```

WAIT OG CAM LIVESTREAMS WORKING? THEY FIGURED IT OUT?

**Comment by roger-** [comment]
@Simcha-Orenbuch 

Probably not:

> 2023-12-20 12:15:55 [WyzeBridge] [GW_GC1] Back Yard Cam not supported

**Comment by cybersholt** [comment]
I was able to connect to the host and other streams manually via VLC for the OG cam. It just doesn't load in the interface. I've yet to connect frigate to it but will report back shortly with the results of that. 

@Simcha-Orenbuch if you've got an OG try connecting to the HLS feed in the browser, it should work. 

I should also note the OG is running the latest firmware 1.0.71 and latest release of wyze bridge.

**Comment by carTloyal123** [comment]
So, quick update on this: I have been working on an android app bundled in a docker container that implements the native libiotvideo.so and libiotp2pav.so libraries. I have successfully completed the grunt work and given an accessId and accessToken for a corresponding deviceId I can output a YUV video stream. My next step is see how we can bring this stream into the bridge and to package my work into a docker container that simply exposes a websocket api for moving config/video data in and out of the container. I will post a repo link once I get it organized. @mrlt8 let me know what you think!

**Comment by carTloyal123** [comment]
(This was all tested using my video doorbell pro but I will test with other camera models as needed)

**Comment by Simcha-Orenbuch** [comment]
> (This was all tested using my video doorbell pro but I will test with other camera models as needed)

I have an extra OG, I can do some testing with it if needed, but you'd need to walk me through the majority of it, not as experienced as nearly anyone in this forum.

**Comment by talormanda** [comment]
> (This was all tested using my video doorbell pro but I will test with other camera models as needed)

If there's an easy way to test I can. I have doorbell pros, OGs, etc. I'm using portainer ATM for the bridge.

**Comment by mrlt8** [comment]
I will have to try to pickup a gwell cam to test things out and see if I can help!

**Comment by carTloyal123** [comment]
> I will have to try to pickup a gwell cam to test things out and see if I can help!

Awesome thank you. I can also get them to test and am wondering if you are interested in me integrating this work with the bridge or had any idea of how to connect the two? Right now I don't have any of the authentication implemented and generate the accessId and accessToken from the gw_user endpoint using the auth mechanisms from wyze-sdk. I then pass the auth to the new application over a simple websocket connection. Essentially I want the new docker container to be passed accessId, accessToken, and a deviceId and then it outputs raw YUV. This stream could be consumed by the bridge and passed out over rtsp or other. For the user of the bridge it could just be a matter of a config option or environment variable to run the bridge with the additional container. What do you think?

**Comment by carTloyal123** [comment]
Also, thanks for everyone willing to test alpha features and I will update this thread with ways to test in the future as integration develops more.

**Comment by cybersholt** [comment]
I also have a OG that I can help test with if needed as well as 2 local v3's and 2 remote v3's. I'm not familiar with reverse engineering but have compiled and loaded custom firmware for the v2 cams. 

It's a shame these are so different, the little spotlight is cool and the picture quality especially at night is pretty impressive for the price.

**Comment by nbetcher** [comment]
> I have been working on an android app bundled in a docker container that implements the native libiotvideo.so and libiotp2pav.so libraries.

What container project would/are you [be] using? docker-android, redroid? M

While I realize you're just being practical (I would be doing the same, if I had time), wouldn't your proposed attack require at least 1GB (or for speed maybe even 1.5-2GB) of RAM? And the bootup speed be in the realm of at least 30 full seconds?

I guess I'm just curious what you're using to defeat the major drawbacks of using a full-blown Android solution in this context.

> I will post a repo link once I get it organized.

That would be excellent. Please, the sooner the better since people often (I've done it, lol) suddenly go missing and all of their proposed work is lost, never to be heard from again. 😜

Thank you for your hard work!!!

**Comment by carTloyal123** [comment]
I have not been able to run an android emulator in a docker container on my main dev machine which is an m2 macbook pro so that is slowing me down at the moment. I have tried all readily available android emulator docker solutions. As far as I can tell there is no way to avoid using a full android emulator so take it or leave it or come up with a new solution. 

I will post the new repo soon, I have it in progress but its not ready for public consumption.

**Comment by jhansche** [comment]
> As far as I can tell there is no way to avoid using a full android emulator 

Keep in mind you may also be able to release it on the play store, or at least give access to an APK that can be installed manually, to allow a real device to run it. I'm sure I'm not the only one that has a handful of old Android devices lying around 😂 This bridge just needs to be able to connect to the Android bridge, whether emulator or real device. I also have an Android Things dev device.

**Comment by nbetcher** [comment]
> I have not been able to run an android emulator in a docker container on my main dev machine which is an m2 macbook pro so that is slowing me down at the moment.

That's your main problem: Mac M2
If you had an x86 machine you wouldn't need an emulator at all. Redroid doesn't even emulate at all as long as you have x86_64 versions of the library. That would be ideal for most.

I hate to say it, but due to Apple's extreme proprietary nature, Mac OS isn't exactly what most users on here are going to be using, ever.

**Comment by carTloyal123** [comment]
> > As far as I can tell there is no way to avoid using a full android emulator
> 
> Keep in mind you may also be able to release it on the play store, or at least give access to an APK that can be installed manually, to allow a real device to run it. I'm sure I'm not the only one that has a handful of old Android devices lying around 😂 This bridge just needs to be able to connect to the Android bridge, whether emulator or real device. I also have an Android Things dev device.

I have no intention of releasing it to the play store. If you want to use it on a spare android device then you can load it manually. 

> > I have not been able to run an android emulator in a docker container on my main dev machine which is an m2 macbook pro so that is slowing me down at the moment.
> 
> That's your main problem: Mac M2 If you had an x86 machine you wouldn't need an emulator at all. Redroid doesn't even emulate at all as long as you have x86_64 versions of the library. That would be ideal for most.
> 
> I hate to say it, but due to Apple's extreme proprietary nature, Mac OS isn't exactly what most users on here are going to be using, ever.

I know how platform architectures work and was not asking for a recitation of my problem but feel free to offer a solution if you wish to comment further. The problem is not apple silicone, but the lack of support from androids command line tools which enable you to automate an emulator on any ARM based device. We also do not have x86 versions for the libraries at all. They are also JNI libraries meaning you basically have to use a real device or emulator to get the JVM environment correct. Again, solutions are welcome but rather not more problems or issues already stated

**Comment by felipel1292** [comment]
Are there any updates available? I have 2 OG cams and would be happy to help with the testing.

**Comment by carTloyal123** [comment]
[Cryze](https://github.com/carTloyal123/cryze) is my project for accessing Wyze Gwell based cameras like the OG Cam, Video Doorbell Pro and I am sure there are others. I have a working prototype and am currently working on a full integration with other projects like this one. Feel free to check that repo periodically as I iron out the kinks and get things release ready.

**Comment by dinan5** [comment]
Awesome news! Thank you!

**Comment by talormanda** [comment]
> [Cryze](https://github.com/carTloyal123/cryze) is my project for accessing Wyze Gwell based cameras like the OG Cam, Video Doorbell Pro and I am sure there are others. I have a working prototype and am currently working on a full integration with other projects like this one. Feel free to check that repo periodically as I iron out the kinks and get things release ready.

Can you add some documentation on how to install this? I have docker / portainer but not super familiar on adding things on my own yet.

**Comment by quistuipater** [comment]
Are the Wyze Cam Pan v3 cameras Gwell based?

[mg]

**Comment by carTloyal123** [comment]
@talormanda as mentioned, I am in the process of getting the project ready for full release. Once that is done, check back to the Cryze repo for install instructions.

@quistuipater do you know the model number of the OG cam? Or have the mac address of one? I think they all start with "GW_"

**Comment by jhansche** [comment]
@quistuipater 
> Are the Wyze Cam Pan v3 cameras Gwell based?

No, the Pan v3 works with this bridge, it is *not* Gwell

**Comment by quistuipater** [comment]
Gracias.

[mg]

**Comment by Simcha-Orenbuch** [comment]
> @talormanda as mentioned, I am in the process of getting the project ready for full release. Once that is done, check back to the Cryze repo for install instructions.
> 
> @quistuipater do you know the model number of the OG cam? Or have the mac address of one? I think they all start with "GW_"
> 
> Edit: yes, the OG and doorbell pro both have the GW model number so it should be safe to assume they are both Gwell Based at the very least.

Do you still need a MAC address or you got the info you needed?

**Comment by carTloyal123** [comment]
> > @talormanda as mentioned, I am in the process of getting the project ready for full release. Once that is done, check back to the Cryze repo for install instructions.
> 
> > 
> 
> > @quistuipater do you know the model number of the OG cam? Or have the mac address of one? I think they all start with "GW_"
> 
> > 
> 
> > Edit: yes, the OG and doorbell pro both have the GW model number so it should be safe to assume they are both Gwell Based at the very least.
> 
> 
> 
> Do you still need a MAC address or you got the info you needed?
> 
> 

I am good, I had mis-read that and I forgot about the list of devices on the home page. Thank you!

**Comment by imbkr** [comment]
is there any update on this? is anyone working on to solve this issue?

**Comment by carTloyal123** [comment]
> [Cryze](https://github.com/carTloyal123/cryze) is my project for accessing Wyze Gwell based cameras like the OG Cam, Video Doorbell Pro and I am sure there are others. I have a working prototype and am currently working on a full integration with other projects like this one. Feel free to check that repo periodically as I iron out the kinks and get things release ready.

Please read the entire thread before posting questions. I have a project that addresses the OG cam and Video Doorbell Pro V1 reference above. @imbkr

**Comment by nbetcher** [comment]
> > [Cryze](https://github.com/carTloyal123/cryze) is my project for accessing Wyze Gwell based cameras like the OG Cam, Video Doorbell Pro and I am sure there are others. I have a working prototype and am currently working on a full integration with other projects like this one. Feel free to check that repo periodically as I iron out the kinks and get things release ready.
> 
> Please read the entire thread before posting questions. I have a project that addresses the OG cam and Video Doorbell Pro V1 reference above. @imbkr

In some fairness it doesn't appear the documentation really ties it all together. E.g. what to do after the APK is  installed in the emulator. (Just to name one)

**Comment by carTloyal123** [comment]
Correct that Cryze is not fully fleshed out as it is a hobby project and is a large undertaking (feel free to clone and make pull requests if you want progress on your own watch. The question was if there was someone/something working towards a solve for this which indeed there is and has been mentioned multiple times in this thread hence my reply. @nbetcher you should not have to worry about the APK at all. If you want clarity on setup please address it to the main [Cryze](https://github.com/carTloyal123/cryze) repo

**Comment by GuySandler** [comment]
has there been a solution in this repo yet?

**Comment by xerootg** [comment]
I've got a hard fork of cryze that is working. It's a bit CPU thirsty but it is actually streaming video to my NVR. I'm activly working on getting hardware acceleration working to try to reduce the footprint. github.com/xerootg/cryze_v2

**Comment by dinan5** [comment]
That's great! thanks.  Will try it.

**Comment by xerootg** [comment]
I'm actively working on it, expect it to both change fundamentally and for there to be issues. I'm likely going to be pulling the raw h264 frames out of android into a third container to handle the RTSP side, possibly today and likely by tomorrow. The CPU consumption of Android's MediaCodec without hardware acceleration is absolutely mind boggling and I really don't feel like recompiling android for the possibility that it might be slightly more performant. I'm using just over 1 core per camera on an i7-12650H which I find totally unacceptable. I did a spike where I rendered the raw output into a MJPEG stream and that was actually worse than double encoding like it does now.

**Comment by xerootg** [comment]
Alright, the raw streamer in my cryze fork seems pretty solid. its been running for a couple days. I got it to stream the raw h264 packets to a tcp client, like ffmpeg, so you'll need to use an intermediary to use my fork like go2rtc (which I am using with frigate). I seriously doubt the gwell cameras will ever be supported outside of an android emulator, as tencent's library uses message queueing to a proprietary backend to get the stream contents. I'm probably going to get a different doorbell after this experience, I'm disappointed in Wyze. The whole gwell API and integration of the gwell products into the wyze platform screams outsourcing chaos.

**Comment by carTloyal123** [comment]
> Alright, the raw streamer in my cryze fork seems pretty solid. its been running for a couple days. I got it to stream the raw h264 packets to a tcp client, like ffmpeg, so you'll need to use an intermediary to use my fork like go2rtc (which I am using with frigate). I seriously doubt the gwell cameras will ever be supported outside of an android emulator, as tencent's library uses message queueing to a proprietary backend to get the stream contents. I'm probably going to get a different doorbell after this experience, I'm disappointed in Wyze. The whole gwell API and integration of the gwell products into the wyze platform screams outsourcing chaos.

Yeah, this is great work. And this is as far as I ever wanted to take it either for the exact same reasons that you mentioned. My whole goal was to use my video doorbell pro in HomeKit, but now that there are cheaper HomeKit secure video enabled doorbell cameras, I would much rather get one of those instead of mess with the horrendous software stack that Wyze is using. Can you share what pipeline you were using for ffmpeg? @xerootg

**Comment by talormanda** [comment]
This is exactly why I started going to tapo and reolink. If I hadn't gotten my doorbell pros for free, I wouldn't be using them as the current option. As nice as wyze is as a company, I don't think their development team knows anything about what's going on.....and this is really evident because anytime you would suggest a really simple feature, it was met with strong backlash about how they can't do it. They are definitely outsourcing.

**Comment by xerootg** [comment]
Thanks @carTloyal123 !! Maybe you'll get some use out of it anyway ;)

Well, all that said, I'm almost done stripping the gwell/iotvideosdk down to the minimum needed to maintain a stream. I've still got some little nags around stalled video streams and token renewal, but it's quite obvious the stack wyze used on these cameras was not designed for long-running stream reading. I'm not going to put effort into transcoding on the container unless someone needs it, it's quite difficult to do hardware interop into a container that doesn't even share a common libc. So all that to say, if you're using my cryze fork, let me know over in my repo and I'll keep working on stabilizing it for as long as it holds my attention. I *might* even try running the android stack in a stripped down android container to give me a bit more control and use less memory!

@mrlt8 I don't know if you want to keep these gwell camera support issues open. There's a solution now and your solution seems to work well for the TUTK cameras. I can confidently say nobody is going to port the tencent iotvideo API to python anytime soon.

**Comment by carTloyal123** [comment]
@mrlt8 I am fine to close this. People can follow instructions at the [Cryze V2](https://github.com/xerootg/cryze_v2) repo for setting up there gwell based VDP, OG, and any others

**Comment by coltography** [comment]
So the solution is to just use a completely separate docker container to use the zoom/og cams?

**Comment by xerootg** [comment]
sadly, yes. and at that, I got upset at the gwell based cameras I had and replaced them all with tapo which have native rtsp. I've not had the time to update the SDK to a more recent version, so cryze v2 is going to be hit and miss for you unless you want to spend some time rewriting the JNI.

**Comment by carTloyal123** [comment]
> So the solution is to just use a completely separate docker container to use the zoom/og cams?

If you are a technical person and want to explore reverse engineering the Gwell iotvideo SDK to enable some sort of direct camera control then go for it. Otherwise Cryze V2 will be the best option, and it's really pretty good considering there was no other way to access these cams previously.

**Comment by coltography** [comment]
Got it, I wasn't fully understanding theres limitation to these specific cameras. How ridiculous. Returned mine!

**Comment by williamjpoole** [comment]
I have installed the Docker Wyze Bridge with the API provided by Wyze. I can detect the camera, but the feed is just a circle of death.

**Comment by XscarfX** [comment]
Just picked up 3 OG cameras has there been any updates to getting this to work with home assistant properly I get everything installed but I get a still image that updates whenever it decides to sometimes after 5 minutes sometimes after 12 hours live streams do not work

**Comment by jhansche** [comment]
No, the OG cams do not with with DWB. Your best option is @carTloyal123's solution described [here](https://github.com/mrlt8/docker-wyze-bridge/issues/677#issuecomment-2371416017).
